client/extensions/zoninator/state/zones/reducer.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import { combineReducers, createReducer } from 'state/utils';
+import {
+	ZONINATOR_FETCH_ERROR,
+	ZONINATOR_FETCH_ZONES,
+	ZONINATOR_UPDATE_ZONES
+} from '../action-types';
+
+export const fetching = createReducer( {}, {
+	[ ZONINATOR_FETCH_ZONES ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: true } ),
+	[ ZONINATOR_UPDATE_ZONES ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
+	[ ZONINATOR_FETCH_ERROR ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
+} );
```

Comment:
Do we need `fetching` state if we use `data-layer`? We're lately trying to avoid that kind of state...